File: 1528-kids-with-the-greatest-number-of-candies/1528-kids-with-the-greatest-number-of-candies.py
class Solution(object):
    def kidsWithCandies(self, candies, extraCandies):
        """
        :type candies: List[int]
        :type extraCandies: int
        :rtype: List[bool]
        """
        n = max(candies)
        res = [False]*len(candies)
        for i in range(len(candies)):
            if candies[i] + extraCandies >= n :
                res[i] = True
            else:
                res[i] = False
        return res

File: 1528-kids-with-the-greatest-number-of-candies/1528-kids-with-the-greatest-number-of-candies.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        List<Boolean> l = new ArrayList<>(candies.length);
        int max = max(candies);
        for (int i = 0;i<candies.length;i++){
            if (candies[i] + extraCandies >= max){
                l.add(true);
            }
            else{
                l.add(false);
            }
        }
        return l;
    }
    private int max(int[] arr)
    {
        int max = arr[0];
        for (int i = 1; i < arr.length; i++)if (arr[i] > max)max = arr[i];
        return max;
    }  
}